pennylane/templates/embeddings.py

```diff
@@ -17,128 +17,138 @@
 used at the beginning of a circuit.
 """
 #pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
-from collections.abc import Iterable
+from pennylane import numpy as np
+from pennylane.qnode import Variable
 from pennylane.ops import RX, RY, RZ, BasisState, Squeezing, Displacement, QubitStateVector
-from pennylane.variable import Variable
-import numpy as np
+from pennylane.templates.utils import (_check_shape, _check_no_variable, _check_wires,
+                                       _check_hyperp_is_in_options, _check_type,
+                                       _get_shape)
 
+TOLERANCE = 1e-3
 
```

Comment:
This file is terrible to read in diff mode due to the change history, I recommend looking at the code...